### PR TITLE
Detect new containers on logs --follow

### PR DESCRIPTION
**Pull request by ulyssessouza**
**What I did**
Detect new containers on `logs --follow`.

**Related issue**
Fixes https://github.com/docker/compose-cli/issues/1635

### CODE REVIEW

File: pkg/compose/logs.go [hunk 1]
```diff
@@ -20,55 +20,83 @@ import (
...
+	"github.com/docker/docker/pkg/stdcopy"
 )
 
 func (s *composeService) Logs(ctx context.Context, projectName string, consumer api.LogConsumer, options api.LogOptions) error {
 	list, err := s.getContainers(ctx, projectName, oneOffExclude, true, options.Services...)
+	if err != nil {
+		return err
+	}
 
+	project, err := s.projectFromContainerLabels(list, projectName)
```
Comment (ndeloof) [inline pkg/compose/logs.go:38]: ? should not be necessary

File: pkg/compose/logs.go [hunk 2]
```diff
@@ -20,55 +20,83 @@ import (
...
+
+	printer := newLogPrinter(consumer)
+
 	eg, ctx := errgroup.WithContext(ctx)
+	eg.Go(func() error {
+		_, err := printer.Run(false, "", nil)
+		return err
+	})
+
+	attached, err := s.attach(ctx, project, printer.HandleEvent, project.ServiceNames())
```
Comment (ndeloof) [inline pkg/compose/logs.go:51]: `logs` should not rely on ContainerAttach but only on ContainerLogs

File: pkg/compose/logs.go [hunk 3]
```diff
@@ -20,55 +20,85 @@ import (
...
 		eg.Go(func() error {
-			r, err := s.apiClient.ContainerLogs(ctx, container.ID, types.ContainerLogsOptions{
-				ShowStdout: true,
-				ShowStderr: true,
-				Follow:     options.Follow,
-				Since:      options.Since,
-				Until:      options.Until,
-				Tail:       options.Tail,
-				Timestamps: options.Timestamps,
+			return s.watchContainers(project, project.ServiceNames(), printer.HandleEvent, containers, func(c types.Container) error {
```
Comment (ndeloof) [inline pkg/compose/logs.go:58]: project.ServiceNames() -> options.Services

you don't (and we don't want you) need to rebuild project model to collect logs

**Comment by mat007** [comment]
😁 
**Conflicting files**
pkg/compose/logs.go

**Pull request merged**